The bookmark code in mate-control-center 1.26.0 requires GLib 2.66
but (a) doesn't bother to check for it, and (b) Tribblix doesn't
have it yet, so revert the g_bookmark_file_set_modified to
g_bookmark_file_set_modified_date_time changes.

--- mate-control-center-1.26.0/libslab/bookmark-agent.c	Sat Aug  7 13:01:13 2021
+++ mate-control-center-1.24.2/libslab/bookmark-agent.c	Wed Mar 24 09:20:22 2021
@@ -112,6 +112,8 @@
                               GFileMonitorEvent, gpointer);
 static void weak_destroy_cb  (gpointer, GObject *);
 
+static gint recent_item_mru_comp_func (gconstpointer a, gconstpointer b);
+
 static gchar *find_package_data_file (const gchar *filename);
 
 static gint BookmarkAgent_private_offset;
@@ -168,7 +186,7 @@
 	g_bookmark_file_set_mime_type (priv->store, item->uri, item->mime_type);
 
 	if (item->mtime)
-		g_bookmark_file_set_modified_date_time (priv->store, item->uri, item->mtime);
+		g_bookmark_file_set_modified (priv->store, item->uri, item->mtime);
 
 	if (item->title)
 		g_bookmark_file_set_title (priv->store, item->uri, item->title);
@@ -321,13 +339,13 @@
 
 			item->uri       = g_strdup (uris [i]);
 			item->mime_type = g_bookmark_file_get_mime_type (store, uris [i], NULL);
-			item->mtime     = g_bookmark_file_get_modified_date_time  (store, uris [i], NULL);
+			item->mtime     = g_bookmark_file_get_modified  (store, uris [i], NULL);
 
 			items_ordered = g_list_prepend (items_ordered, item);
 		}
 	}
 
-	items_ordered = g_list_sort (items_ordered, g_date_time_compare);
+	items_ordered = g_list_sort (items_ordered, recent_item_mru_comp_func);
 
 	g_strfreev (uris);
 
@@ -356,7 +374,7 @@
 		item = (BookmarkItem *) node->data;
 
 		g_bookmark_file_set_mime_type (priv->store, item->uri, item->mime_type);
-		g_bookmark_file_set_modified_date_time  (priv->store, item->uri, item->mtime);
+		g_bookmark_file_set_modified  (priv->store, item->uri, item->mtime);
 
 		bookmark_item_free (item);
 	}
@@ -718,7 +720,7 @@
 			priv->items [i]->uri       = g_strdup (uris_ordered [i]);
 			priv->items [i]->title     = g_bookmark_file_get_title     (priv->store, uris_ordered [i], NULL);
 			priv->items [i]->mime_type = g_bookmark_file_get_mime_type (priv->store, uris_ordered [i], NULL);
-			priv->items [i]->mtime     = g_bookmark_file_get_modified_date_time (priv->store, uris_ordered [i], NULL);
+			priv->items [i]->mtime     = g_bookmark_file_get_modified  (priv->store, uris_ordered [i], NULL);
 			priv->items [i]->app_name  = NULL;
 			priv->items [i]->app_exec  = NULL;
 
@@ -1223,4 +1225,10 @@
 weak_destroy_cb (gpointer data, GObject *g_obj)
 {
 	instances [GPOINTER_TO_INT (data)] = NULL;
+}
+
+static gint
+recent_item_mru_comp_func (gconstpointer a, gconstpointer b)
+{
+	return ((BookmarkItem *) b)->mtime - ((BookmarkItem *) a)->mtime;
 }
--- mate-control-center-1.26.0/libslab/bookmark-agent.h	Sat Aug  7 13:01:13 2021
+++ mate-control-center-1.24.2/libslab/bookmark-agent.h	Wed Mar 24 09:20:22 2021
@@ -22,10 +22,11 @@
 #define __BOOKMARK_AGENT_H__
 
 #include <time.h>
-#include <glib.h>
 #include <glib-object.h>
 
-G_BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #define BOOKMARK_AGENT_TYPE         (bookmark_agent_get_type ())
 #define BOOKMARK_AGENT(o)           (G_TYPE_CHECK_INSTANCE_CAST ((o), BOOKMARK_AGENT_TYPE, BookmarkAgent))
@@ -41,7 +42,7 @@
 	gchar  *uri;
 	gchar  *title;
 	gchar  *mime_type;
-	GDateTime  *mtime;
+	time_t  mtime;
 	gchar  *icon;
 	gchar  *app_name;
 	gchar  *app_exec;
@@ -86,6 +87,8 @@
 
 void           bookmark_item_free           (BookmarkItem *item);
 
-G_END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
-#endif /* __BOOKMARK_AGENT_H__ */
+#endif
